### ISSUE TITLE
AI project

### ISSUE THREAD

**Issue by mohammadmehdizare** [mohammadmehdizare/AI-project#1]
import tensorflow as tf
from tensorflow.keras import layers, models
from tensorflow.keras.preprocessing.image import ImageDataGenerator

# مرحله 1: پیش‌پردازش داده‌ها
train_datagen = ImageDataGenerator(
    rescale=1./255, 
    rotation_range=40,
    width_shift_range=0.2,
    height_shift_range=0.2,
    shear_range=0.2,
    zoom_range=0.2,
    horizontal_flip=True,
)

test_datagen = ImageDataGenerator(rescale=1./255)

train_generator = train_datagen.flow_from_directory(
    'data/train',
    target_size=(150, 150),
    batch_size=32,
    class_mode='binary'  # یا 'categorical' در صورت داشتن چند کلاس
)

validation_generator = test_datagen.flow_from_directory(
    'data/validation',
    target_size=(150, 150),
    batch_size=32,
    class_mode='binary'  # یا 'categorical' در صورت داشتن چند کلاس
)

# مرحله 2: ساخت مدل
model = models.Sequential([
    layers.Conv2D(32, (3, 3), activation='relu', input_shape=(150, 150, 3)),
    layers.MaxPooling2D(2, 2),
    
    layers.Conv2D(64, (3, 3), activation='relu'),
    layers.MaxPooling2D(2, 2),
    
    layers.Conv2D(128, (3, 3), activation='relu'),
    layers.MaxPooling2D(2, 2),
    
    layers.Conv2D(128, (3, 3), activation='relu'),
    layers.MaxPooling2D(2, 2),
    
    layers.Flatten(),
    layers.Dense(512, activation='relu'),
    layers.Dense(1, activation='sigmoid')  # تغییر به 'softmax' در صورت داشتن چند کلاس
])

# کامپایل مدل
model.compile(loss='binary_crossentropy',  # یا 'categorical_crossentropy'
              optimizer='adam',
              metrics=['accuracy'])

# مرحله 3: آموزش مدل
model.fit(train_generator, 
          steps_per_epoch=train_generator.samples // train_generator.batch_size,
          validation_data=validation_generator, 
          validation_steps=validation_generator.samples // validation_generator.batch_size,
          epochs=15)

# مرحله 4: ارزیابی مدل
model.evaluate(validation_generator)
